### PR TITLE
Signing CD Action

### PR DESCRIPTION
**Pull request by jasonleenaylor**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/flexbridge/397)

### CODE REVIEW

File: src/LibFLExBridge-ChorusPluginTests/LibFLExBridge-ChorusPluginTests.csproj [hunk 1]
```diff
@@ -12,7 +12,7 @@
     <PackageReference Include="GitVersion.MsBuild" Version="5.10.3" PrivateAssets="all" />
     <PackageReference Include="NUnit" Version="3.13.3" />
     <PackageReference Include="SIL.Chorus.LibChorus.TestUtilities" Version="$(ChorusVersion)" />
-    <PackageReference Include="SIL.Chorus.Mercurial" Version="6.5.1.25" IncludeAssets="build" />
+    <PackageReference Include="SIL.Chorus.Mercurial" Version="6.5.1.22" IncludeAssets="build" />
```
Comment (rmunn) [inline src/LibFLExBridge-ChorusPluginTests/LibFLExBridge-ChorusPluginTests.csproj]: Why this downgrade? SIL.Chorus.Mercurial 6.5.1.25 fixed running Mercurial on Linux under Python 3, and AFAIK made no other changes. Is there something that I broke in creating 6.5.1.25 that was working correctly with 6.5.1.22 which made the downgrade necessary?

**Comment by megahirt** [comment]
This is a monumental effort, @jasonleenaylor nice work!  I assume you are happy for this to get merged in with a review and not wait for a signing service to be developed.

**Comment by jasonleenaylor** [comment]
Yes @megahirt I would like to get this merged to use until we have a signing service.

**Comment by jasonleenaylor** [comment]
From Ken Zook:
"It installed. I was able to get project from colleague from LD, and did a S/R project to LD and it worked. I think we have a winner. It's version FLEx Bridge 4.2.0-alpha.20 13/Jun/2024. It has mercurial 6.5.1 installed. Looks like it's ready for the test team to give it a good workout. "

I would like to get some PRs in the codesign repo merged and update this PR with a real tagged version but otherwise it is ready.
https://github.com/sillsdev/codesign/pull/18
https://github.com/sillsdev/codesign/pull/19

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -5,16 +5,23 @@ on:
...
   cancel-in-progress: true
 
 jobs:
-  windows_debug_build_and_test:
+  build_and_test:
     env:
       CROWDIN_API_KEY: ${{ secrets.CROWDIN_API_KEY }}
-    name: Build Debug and run Tests
+      FILESTOSIGNLATER: "${{ github.workspace }}\\filesToSign"
+      teamcity_build_checkoutDir: ${{ github.workspace }}
```
Comment (megahirt) [inline .github/workflows/ci-cd.yml:23]: Should this variable be renamed?

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -30,58 +37,292 @@ jobs:
...
-          dotnet-version: | 
-            2.1.x
-            3.1.x
-            5.0.x
+          dotnet-version:
+            8.0.x
         if: github.event_name != 'pull_request'
-        
+
+      - name: Downgrade Wix Toolset - remove when runner has 3.14.2
```
Comment (megahirt) [inline .github/workflows/ci-cd.yml:56]:
```suggestion
      - name: Downgrade Wix Toolset - remove when runner has 3.14.2
      # See: https://github.com/actions/runner-images/issues/9667
```

**Review by megahirt — APPROVED** [review]
LGTM

**Review by rmunn — APPROVED** [review]
I have no experience with WiX so if there are any mistakes in that part of the PR then I won't have caught them. But the GHA workflow changes look good to me.

File: .github/workflows/ci-cd.yml (hunk 2 above)
Comment (rmunn) [reply .github/workflows/ci-cd.yml:23]: It's used by FLExBridge.proj to get the root directory of the project. We should keep it until the TeamCity build is finally gone, then we can rename it both here and in FLExBridge.proj.

File: .github/workflows/ci-cd.yml (hunk 3 above)
Comment (rmunn) [reply .github/workflows/ci-cd.yml:56]: I'd rather put that in a comment rather than make an already-long step name even longer.

Comment (megahirt) [reply .github/workflows/ci-cd.yml:56]: good call.  I moved it to the next line as a comment

**Review by jasonleenaylor — COMMENTED** [review]
> Dismissed @rmunn from a discussion.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/flexbridge/397)* status: 0 of 8 files reviewed, 2 unresolved discussions (waiting on @megahirt and @rmunn)

___
*[`src/LibFLExBridge-ChorusPluginTests/LibFLExBridge-ChorusPluginTests.csproj` line 15 at r1](https://reviewable.io/reviews/sillsdev/flexbridge/397#gh-1570047837:-O09KjGOCDvu9kwBRLpY:b-896fix) ([raw file](https://github.com/sillsdev/flexbridge/blob/53b43436ac680421348b4331b7d724bda73d7e72/src/LibFLExBridge-ChorusPluginTests/LibFLExBridge-ChorusPluginTests.csproj#L15)):*
<details><summary><i>Previously, rmunn (Robin Munn) wrote…</i></summary><blockquote>

Why this downgrade? SIL.Chorus.Mercurial 6.5.1.25 fixed running Mercurial on Linux under Python 3, and AFAIK made no other changes. Is there something that I broke in creating 6.5.1.25 that was working correctly with 6.5.1.22 which made the downgrade necessary?
</blockquote></details>

Done.

**Review by jasonleenaylor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/flexbridge/397)* status: 0 of 8 files reviewed, 2 unresolved discussions (waiting on @megahirt and @rmunn)

___
*[`.github/workflows/ci-cd.yml` line 56 at r4](https://reviewable.io/reviews/sillsdev/flexbridge/397#gh-1650427542:-O09SyekEPNYE2FnVDFa:b-896fix) ([raw file](https://github.com/sillsdev/flexbridge/blob/79893d711bc5b72c42263dbbc3d451dbe3488c86/.github/workflows/ci-cd.yml#L56)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

good call.  I moved it to the next line as a comment
</blockquote></details>

Done.

**Review by jasonleenaylor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/flexbridge/397)* status: 0 of 8 files reviewed, 2 unresolved discussions (waiting on @megahirt and @rmunn)

___
*[`.github/workflows/ci-cd.yml` line 23 at r4](https://reviewable.io/reviews/sillsdev/flexbridge/397#gh-1650427154:-O09T0O13wlI7NPDSyxK:b-896fix) ([raw file](https://github.com/sillsdev/flexbridge/blob/79893d711bc5b72c42263dbbc3d451dbe3488c86/.github/workflows/ci-cd.yml#L23)):*
<details><summary><i>Previously, rmunn (Robin Munn) wrote…</i></summary><blockquote>

It's used by FLExBridge.proj to get the root directory of the project. We should keep it until the TeamCity build is finally gone, then we can rename it both here and in FLExBridge.proj.
</blockquote></details>

Done.

**Pull request merged**